homeassistant/components/zwave_js/__init__.py

```diff
@@ -378,22 +401,28 @@ async def disconnect_client(
 
 async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Unload a config entry."""
-    unload_ok = all(
-        await asyncio.gather(
-            *[
+    for unsub in hass.data[DOMAIN][entry.entry_id][DATA_UNSUBSCRIBE]:
+        unsub()
+
+    tasks = []
+    for platform, task in hass.data[DOMAIN][entry.entry_id][
+        DATA_PLATFORM_SETUP
+    ].items():
+        if task.done():
+            tasks.append(
                 hass.config_entries.async_forward_entry_unload(entry, platform)
-                for platform in PLATFORMS
-            ]
-        )
-    )
+            )
+        else:
+            task.cancel()
+            tasks.append(task)
+
+    unload_ok = all(await asyncio.gather(*tasks))
+
     if not unload_ok:
         return False
```

Comment:
Should we still return early here if we have already unsubscribed the node listeners? Maybe we should continue and try to clean up and disconnect as much as possible.

Reply:
I like the approach of trying to clean up as much as possible. Update incoming